File: Server/src/main/content/global/skill/production/cooking/recipe/pie/impl/MeatPie.java
package content.global.skill.production.cooking.recipe.pie.impl;

import content.global.skill.production.cooking.recipe.pie.PieRecipe;
import core.game.interaction.NodeUsageEvent;
import core.game.node.entity.player.Player;
import core.game.node.item.Item;

/**
 * The Meat pie.
 */
public class MeatPie extends PieRecipe {

    private static final Item UNCOOKED_PIE = new Item(2319);

    private static final Item COOKED_MEAT = new Item(2142);

    private static final Item COOKED_CHICKEN = new Item(2140);

    private static final Item COOKED_RABBIT = new Item(3228);

    @Override
    public void mix(final Player player, final NodeUsageEvent event) {
        if (player.getInventory().remove(event.getUsedItem()) && player.getInventory().remove(event.getBaseItem())) {
            player.getInventory().add(getProduct());
            player.getPacketDispatch().sendMessage(getMixMessage(event));
        }
    }

    @Override
    public Item getProduct() {
        return UNCOOKED_PIE;
    }

    @Override
    public Item[] getIngredients() {
        return new Item[]{COOKED_MEAT, COOKED_CHICKEN, COOKED_RABBIT};
    }

    @Override
    public String getMixMessage(final NodeUsageEvent event) {
        return "You fill the pie with meat.";
    }
}

File: Server/src/main/content/global/skill/production/cooking/recipe/pie/impl/MeatPie.kt
package content.global.skill.production.cooking.recipe.pie.impl

import content.global.skill.production.cooking.recipe.pie.PieRecipe
import cfg.consts.Items
import core.api.removeItem
import core.api.sendMessage
import core.game.interaction.NodeUsageEvent
import core.game.node.entity.player.Player
import core.game.node.item.Item

/**
 * Meat pie.
 */
class MeatPie : PieRecipe() {

    override fun mix(player: Player, event: NodeUsageEvent) {
        if (removeItem(player, event.usedItem) && removeItem(player, event.baseItem)) {
            player.inventory.add(product)
            sendMessage(player, getMixMessage(event))
        }
    }

    override fun getProduct(): Item {
        return UNCOOKED_PIE
    }

    override fun getIngredients(): Array<Item> {
        return arrayOf(COOKED_MEAT, COOKED_CHICKEN, COOKED_RABBIT)
    }

    override fun getMixMessage(event: NodeUsageEvent): String {
        return "You fill the pie with meat."
    }

    companion object {
        private val UNCOOKED_PIE = Item(Items.UNCOOKED_MEAT_PIE_2319)
        private val COOKED_MEAT = Item(Items.COOKED_MEAT_2142)
        private val COOKED_CHICKEN = Item(Items.COOKED_CHICKEN_2140)
        private val COOKED_RABBIT = Item(Items.COOKED_RABBIT_3228)
    }
}
